### PR TITLE
[sui-framework] Remove "delegate" and replace with `stake`

### PR DESCRIPTION
**Pull request by tzakian**
This does a large rename of "delegation/delegate" (etc) to the equivalent "stake" name, e.g., `request_add_delegation` becomes `request_add_stake`.

### Entry functions renamed (all within `sui_system.move`):
```
request_add_delegation ~> request_add_stake
request_add_delegation_mul_coin ~> request_add_stake_mul_coin
request_add_delegation_with_locked_coin ~> request_add_stake_with_locked_coin
request_add_delegation_with_mul_locked_coin ~> request_add_stake_with_mul_locked_coin
request_withdraw_delegation ~> request_withdraw_stake
```

### Move Object fields

```
StakingPool::pending_delegation ~> pending_stake
StakedSui::delegation_activation_epoch ~> stake_activation_epoch 
```

### Sui types

This renames a couple sui types since they no longer made sense to be called what they were:

```
DelegatedStake ~> Stake
DelegationStatus ~> StakeStatus
```

This then performs the core set of renamings to make the system happy with this. So there are still a number of functions in Rust and Typescript that will contain their old delegation name.

Note that on the typescript side we keep DelegatedStake since (1) there was already a type called Stake (which meant something else) and in the context that I could see it being used it appeared as though DelegatedStake really did mean "this is my stake that I have delegated to this validator." 

## Test Plan 

Make sure existing tests still pass.

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [X] user-visible impact
- [X] breaking change for a client SDKs
- [X] breaking change for FNs (FN binary must upgrade)
- [X] breaking change for validators or node operators (must upgrade binaries)
- [X] breaking change for on-chain data layout
- [X] necessitate either a data wipe or data migration

### CODE REVIEW

**Comment by patrickkuo** [comment]
the governance api PR https://github.com/MystenLabs/sui/pull/8848 has some of the same renaming changes on the RPC layer, SDK and the FE apps, this will create quite a bit of conflicts, can we put this on hold 🙏?

File: crates/sui-framework/sources/governance/sui_system.move
Comment (Jordan-Mysten) [inline crates/sui-framework/sources/governance/sui_system.move:308]: We should really just remove this method.

Comment (tzakian) [reply crates/sui-framework/sources/governance/sui_system.move:308]: To keep things clean I might leave this in for now so this is a strict renaming pass, but will do a fast follow to remove this.

**Comment by tzakian** [comment]
The plan is to wait for #9058 to land, and once it does this PR will be rebased on that and then landed. 

In the meantime please review and let me know if there is anywhere I have missed or that you think an error was made.

File: crates/sui-framework/sources/governance/sui_system.move
Comment (emmazzz) [reply crates/sui-framework/sources/governance/sui_system.move:313]: But I believe this function is used in our `sui-transaction-builder` as well as `sui-rosetta`:
https://github.com/MystenLabs/sui/blob/ffc34df782a597e62e8191b934fa752863bdb376/crates/sui-transaction-builder/src/lib.rs#L663
cc @patrickkuo

File: crates/sui-framework/sources/governance/sui_system.move
Comment (emmazzz) [inline crates/sui-framework/sources/governance/sui_system.move:581]: stake stake 🤣 but let's land the PR first and fix this later

File: crates/sui-framework/tests/safe_tests.move
Comment (emmazzz) [inline crates/sui-framework/tests/safe_tests.move:26]: Hmm I don't think this file should be changed. Here we actually want to use the word "delegate" because we are delegating the capability to transfer money to someone else.

Comment (tzakian) [reply crates/sui-framework/tests/safe_tests.move:26]: Will fix :)

**Review by emmazzz — APPROVED** [review]
We need to revert the changes to one file but other than that LGTM, approving to unblock. Thank you so much! @tzakian

File: crates/sui-framework/sources/governance/sui_system.move
Comment (tzakian) [reply crates/sui-framework/sources/governance/sui_system.move]: Changed this at the same time as I was reverting the `safe_tests` file.

**Pull request merged**
